File: 350-intersection-of-two-arrays-ii/intersection-of-two-arrays-ii.py
class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        nums1.sort()
        nums2.sort()
        i,j=0,0
        n1,n2=len(nums1),len(nums2)
        res=[]
        while i<n1 and j<n2:
            while j<n2 and nums2[j]<nums1[i]:
                j+=1
            if j==n2:
                return res
            if nums1[i]!=nums2[j]:
                i+=1
                continue 
            res.append(nums1[i])
            i+=1
            j+=1
        return res

File: 350-intersection-of-two-arrays-ii/intersection-of-two-arrays-ii.cpp
class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
        int n1 = nums1.size();
        int n2 = nums2.size();

        unordered_map<int, int> map;
        for(int i=0;i<n1;i++){
            map[nums1[i]]++;
        }

        // for(auto it=map.begin();it!=map.end();it++) {
        //     cout<<it->first<<" : "<<it->second<<endl;
        // }
        vector<int> res;
        for(int a:nums2){
            if(!map[a] || map[a]==0)
                continue;
            map[a]--; 
            res.push_back(a);
        }

        return res;
    }
};